src/vs/workbench/contrib/tasks/test/common/taskConfiguration.test.ts

```diff
@@ -1760,3 +1765,107 @@ suite('Bugs / regression tests', () => {
 		testConfiguration(external, builder);
 	});
 });
+
+class TestNamedProblemMatcher implements Partial<ProblemMatcher> {
+}
+
+class TestParseContext implements Partial<ParseContext> {
+}
+
+suite('To task configuration from', () => {
+	let instantiationService: TestInstantiationService;
+	let parseContext: ParseContext;
+	let namedProblemMatcher: NamedProblemMatcher;
+	let problemReporter: ProblemReporter;
+	setup(() => {
+		instantiationService = new TestInstantiationService();
+		namedProblemMatcher = instantiationService.createInstance(TestNamedProblemMatcher);
+		namedProblemMatcher.name = 'real';
+		namedProblemMatcher.label = 'real label';
+		problemReporter = new ProblemReporter();
+		parseContext = instantiationService.createInstance(TestParseContext);
+		parseContext.problemReporter = problemReporter;
+		parseContext.namedProblemMatchers = { 'real': namedProblemMatcher };
+		parseContext.uuidMap = new UUIDMap();
+	});
+	suite('ProblemMatcher config', () => {
+		test('returns [] and an error for an unknown problem matcher', () => {
+			const result = (ProblemMatcherConverter.from('$fake', parseContext));
+			assert.deepEqual(result.value, []);
+			assert.strictEqual(result.errors?.length, 1);
+		});
+		test('returns config for a known problem matcher', () => {
+			const result = (ProblemMatcherConverter.from('$real', parseContext));
+			assert.strictEqual(result.errors?.length, 0);
+			assert.deepEqual(result.value, [{ "label": "real label" }]);
+		});
+		test('returns config for a known problem matcher including applyTo', () => {
+			namedProblemMatcher.applyTo = ApplyToKind.closedDocuments;
+			const result = (ProblemMatcherConverter.from('$real', parseContext));
+			assert.strictEqual(result.errors?.length, 0);
+			assert.deepEqual(result.value, [{ "label": "real label", "applyTo": ApplyToKind.closedDocuments }]);
+		});
+	});
+	suite('TaskParser external config', () => {
+		suite('CustomTask', () => {
+			suite('incomplete config reports an appropriate error for missing', () => {
+				test('name', () => {
+					const result = TaskParser.from([{} as CustomTask], {} as Globals, parseContext, {} as TaskConfigSource);
```

Comment:
Are these casts needed? Would it be helpful to have a single test object that implements Globals for example? Even if you're still doing the cast there, at least there's just one then.